lib/dal/src/change_set.rs

```diff
@@ -477,7 +485,17 @@ impl ChangeSet {
         ctx.do_rebase_request(rebase_request).await?;
 
         self.update_status(ctx, ChangeSetStatus::Applied).await?;
+        let user = match ctx.history_actor() {
+            HistoryActor::User(user_pk) => User::get_by_pk(ctx, *user_pk)
+                .await?
+                .ok_or(ChangeSetError::InvalidActor(*user_pk))?,
 
+            HistoryActor::SystemInit => return Err(ChangeSetError::InvalidUserSystemInit),
+        };
```

Comment:
@jobelenus this is what is failing your test. Tests use the `SystemInit` history actor and it's valid for them to do so.